up42/asset.py

```diff
@@ -100,9 +70,39 @@ def stac_items(self) -> pystac.ItemCollection:
                 ],
             },
         }
-        pystac_search = pystac_client_aux.search(filter=stac_search_parameters)
-        resulting_item = pystac_search.get_all_items()
-        return resulting_item
+        pystac_asset_search = pystac_client_aux.search(filter=stac_search_parameters)
+        return (pystac_client_aux, pystac_asset_search)
+
+    @property
+    def stac_info(self) -> Optional[pystac.Collection]:
+        """
+        Gets the storage STAC information for the asset as a FeatureCollection.
+        One asset can contain multiple STAC items (e.g. the pan- and multispectral images).
+        """
+        pystac_client_aux, pystac_asset_search = self._stac_search
+        resulting_item = next(
+            pystac_asset_search.get_item_collections(), None
```

Comment:
This [method is deprecated since version 0.4.0](https://pystac-client.readthedocs.io/en/latest/api.html#pystac_client.ItemSearch.get_item_collections), and since we are not pinning the version, consider using alternative instead (I think it's `item_collection()'.

On related note, I will create a ticket for pinning the dependencies version in the requirements file.

Reply:
Nice catch! ✅